--- linux-2.6.29.6/arch/x86/vdso/Makefile~	2009-07-02 18:41:20.000000000 -0500
+++ linux-2.6.29.6/arch/x86/vdso/Makefile	2013-12-30 13:57:31.714685411 -0600
@@ -69,7 +69,7 @@
 vdso32-images			= $(vdso32.so-y:%=vdso32-%.so)
 
 CPPFLAGS_vdso32.lds = $(CPPFLAGS_vdso.lds)
-VDSO_LDFLAGS_vdso32.lds = -m elf_i386 -Wl,-soname=linux-gate.so.1
+VDSO_LDFLAGS_vdso32.lds = -m32 -Wl,-soname=linux-gate.so.1
 
 # This makes sure the $(obj) subdirectory exists even though vdso32/
 # is not a kbuild sub-make subdirectory.
